### PR TITLE
xmtp-node-js-tools

### PR DESCRIPTION
**Pull request by fabriguespe**
This PR unifies different repos under `xmtp-node-js` to showcase XMTP JS SDK features.

- Enhanced README's
    - [bot-kit-pro](https://github.com/xmtp/bot-kit-pro/blob/xmtpnodejs/packages/bot-kit-pro/README.md)
    - [cli-starter](https://github.com/xmtp/bot-kit-pro/blob/xmtpnodejs/packages/cli-starter/README.md)
    - [bot-starter](https://github.com/xmtp/bot-kit-pro/blob/xmtpnodejs/packages/bot-starter/README.md)
    - [redis-persistence](https://github.com/xmtp/bot-kit-pro/blob/xmtpnodejs/packages/redis-persistence/README.md)
    - [fs-persistence](https://github.com/xmtp/bot-kit-pro/blob/xmtpnodejs/packages/fs-persistence/README.md)
    - [bot-examples](https://github.com/xmtp/bot-kit-pro/blob/xmtpnodejs/packages/bot-examples/README.md)
    - [grpc-api-client](https://github.com/xmtp/bot-kit-pro/blob/xmtpnodejs/packages/grpc-api-client/README.md)
- Added [`cli-starter`](https://github.com/xmtp/cli-starter)
  -  Completed example flow. Working successfully
  - Added test
  - All commands
    ```jsx
    ./xmtp init
    ./xmtp send 0xF8cd371Ae43e1A6a9bafBB4FD48707607D24aE43 "Hello world"
    ./xmtp list-messages 0xF8cd371Ae43e1A6a9bafBB4FD48707607D24aE43
    ./xmtp stream-all
    ./xmtp stream 0xF8cd371Ae43e1A6a9bafBB4FD48707607D24aE43
    ```
- Added [xmtp-bot-starter](https://github.com/xmtp/xmtp-bot-starter)
  - Improved bot-starter typescript snippets

  
  
- Changed to `yarn@4.0.0`
- Extended `tsconfig.base.json` to all packages
- Added `example` folder with JS `demos`
- All checks passed ✅

### CODE REVIEW

**Comment by fabriguespe** [comment]
- Checks passed

File: packages/fs-persistence/README.md [hunk 1]
```diff
@@ -1,15 +1,56 @@
...
+cd packages
+cd fs-persistence
+```
+
+### Install Dependencies and Build
+
+Install the necessary packages:
+
+```bash
+yarn install
```
Comment (neekolas) [inline packages/fs-persistence/README.md:37]: This is useful for someone who wants to build and work on the repo, but I suggest most people just really care about "how do I use this package in my XMTP app?". Maybe we should switch the order around

File: packages/redis-persistence/README.md [hunk 2]
```diff
@@ -1,20 +1,92 @@
...
+
+   > **Tip**  
+   > If you have an earlier version of yarn installed, try running `yarn set version berry` to upgrade to a compatible version.
+
+4. **Docker**: Optional, for containerizing Redis or your application.
+5. **Node.js**: The runtime for your application.
+
+## Installation
+
+### Clone the Repository
```
Comment (neekolas) [inline packages/redis-persistence/README.md:23]: Same comment as above. I think we should start with how to consume the package, and then worry about the small % of users who are actually going to work in this repo

File: packages/grpc-api-client/main.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { GrpcApiClient } from "@xmtp/grpc-api-client"
```
Comment (neekolas) [inline packages/grpc-api-client/main.js:1]: WDYT about renaming this to `example.js` or putting it in an examples folder?

**Comment by neekolas** [comment]
Naming the repo is a bit challenging. We already have `xmtp-node-go`, which is the node software. That makes `xmtp-node-js` sound like a port of Go code to Javascript.

What about `xmtp-node-js-tools` or something? Open to other suggestions

**Review by neekolas — APPROVED** [review]

File: packages/cli-starter/xmtp_wallet [hunk 4]
```diff
@@ -0,0 +1 @@
+farm solve oxygen capital expire original globe stamp blue feed scene injury
```
Comment (daria-github) [inline packages/cli-starter/xmtp_wallet:1]: is this intentionally in the PR?

**Pull request merged**